### PR TITLE
#12463 Allow users to send mails to case, contact, event participant …

### LINKED ISSUE #12463: Allow users to send mails to case, contact, event participant and travel entry persons [5]
### Feature Description

Add a new "Emails" side component to cases, contacts, event participants and travel entries, and allow users to send emails based on email templates to the associated person.

### Added Value/Benefit

Users will be able to send external emails through SORMAS without having to go through an external email client and having to manually fill in information all the time.

### Acceptance Criteria

- [x] A new "Emails" side component has been added to cases, contacts, event participants and travel entries
- [x] A "Send email" button has been added to the component
- [x] When the associated person has no email address specified, a hint is displayed in the component: "Case/contact/event participant/travel entry person has no email address specified" and the button is disabled
- [x] Clicking on the button brings up a popup that allows the user to configure and send the email:
  - [x] A "Choose template" dropdown contains all email templates for the respective entity and allows the user to choose one
  - [x] An "Email address" dropdown contains the primary email address and, if available, all additional email addresses from the person contact details entities associated with the person
  - [x] The primary email address is selected by default and has a `(primary)` suffix; this suffix is translatable through i18n
  - [x] The user can close the popup with a "Discard" button, or send the email with a "Send" button; when the send button is clicked and the template or email address dropdowns are empty, show an error notification and tell the user to select an email template and an email address
- [x] The new components are only visible for users with the `EXTERNAL_EMAIL_SEND` user right introduced in #12466 and when the new `EXTERNAL_EMAILS` feature type introduced in #12461 is enabled
- [x] It's verified that emails sent this way are actually sent

### Mockups

image
image

### Additional Information

Password protection and document attachment will be handled in separate issues.

### PR DESCRIPTION
**Pull request by leventegal-she**
…and travel entry persons

Fixes #12463

### CODE REVIEW

**Comment by sergiupacurariu** [comment]
CI with Maven fails
   ExternalEmailFacadeEjbTest.testSendEmailToCasePerson

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12770

**Comment by sergiupacurariu** [comment]
please rename the email configuration pop-up
image

File: sormas-backend/src/main/java/de/symeda/sormas/backend/docgeneration/DocumentTemplateFacadeEjb.java [hunk 1]
```diff
@@ -462,6 +472,52 @@ private void assertRequredUserRight(DocumentWorkflow documentWorkflow) {
 		}
 	}
 
+	private static void validateEmailTemplate(byte[] document) {
+		EmailTemplateTexts emailTemplateTexts = splitTemplateContent(new String(document, StandardCharsets.UTF_8), false);
+		if (StringUtils.isEmpty(emailTemplateTexts.subject)
+			|| !emailTemplateTexts.subject.startsWith(EMAIL_TEMPLATE_SUBJECT_PREFIX)
+			|| emailTemplateTexts.subject.length() <= EMAIL_TEMPLATE_SUBJECT_PREFIX.length()
```
Comment (sergiupacurariu) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/docgeneration/DocumentTemplateFacadeEjb.java:479]: it could be strictly smaller

**Review by sergiupacurariu — CHANGES_REQUESTED** [review]

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12770

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12770

**Review by sergiupacurariu — APPROVED** [review]

**Pull request merged**